### ISSUE TITLE
Hierarchical tags

### ISSUE THREAD

**Issue by GondekNP** [olgaliak/active-learning-detect#31]
Abram and I were discussing a few edge cases that we are running into, specifically when we can identify a blurry or cut-off image down to a general group but not down to a species (for eg, can tell it's a mammal but can't tell which, or when we have a bad angle on a shearwater/petrel and cannot determine which), and we were wondering about the possibility of having hierarchical classes. In our biological use-case, this makes a lot of sense - for example, if we label something as a hawaiian petrel, we can assume it's an animal and part of the Procellariiformes order, part of the Pterodroma genus, etc etc. Then, if we couldn't ID a bird down to species, we could simply label it with the most specific label we have for it, whether it's Pterodroma, Procellariiformes, or just Aves. 

I'm not sure how much re-invention would be required to make that work. In a perfect world, the model would be able to use all of the levels of the tag hierarchy, meaning that all of the specific tags would also be incorporated in each more general class, instead of each level being considered its own unique class. 

Just curious if this is possible!

**Comment by abfleishman** [comment]
I think this is a really interesting idea.  The model might not be great at identifying a specific type of object, the `model of a car` for example, but it knows that it is a type of `car`, which is a type of `vehicle`, which is a type of `manmade object`.

**Comment by abfleishman** [comment]
I think that this would be pretty cool!  Maybe a model/project would need a "tag library" file that defines the hierarchy and how tags are nested within each other.   This would be supplied upon project initialization and could be updated/added to as new tags were needed for unanticipated labels.

**Comment by yashpande** [comment]
Yeah, it could probably just be a list of child -> parent pairs and the code could create the tree structure.
